piyushgarg-dev/leetcode-whisper-chrome-extension
# feat: Add theme support and extension icons

This PR adds theme support to the LeetCode Whisper Chrome extension, allowing users to switch between light and dark modes.

![image](https://github.com/user-attachments/assets/0549334b-dc7a-4ed8-9e02-ad81aacc115b)
![image](https://github.com/user-attachments/assets/d8aa0227-d77b-47bf-9310-ace159b7ebd3)


### 🎨 Features Added
- Theme toggle functionality with light/dark mode support
- Persistent theme storage using localStorage
- System theme detection
- Smooth theme transitions
- New extension icons in multiple sizes (16x16, 32x32, 48x48, 128x128)

### 🔧 Technical Changes
1. New Components:
   - `ThemeProvider`: Manages theme state and context
   - `ThemeToggle`: UI component for switching themes
   
2. Modified Files:
   - `src/App.tsx`: Added theme-aware styling and theme toggle
   - `src/main.tsx`: Wrapped app with ThemeProvider
   - `manifest.json`: Updated with new icon configurations
   - `tsconfig.app.json`: Updated TypeScript configuration

3. UI/UX Improvements:
   - Theme-aware styling using Tailwind CSS
   - Improved color contrast and readability
   - Consistent UI across light/dark modes

### 📝 Testing Instructions
1. Load the extension in Chrome
2. Verify theme toggle functionality
3. Check theme persistence after browser restart
4. Verify system theme detection
5. Check icon rendering in different contexts:
   - Extension toolbar
   - Chrome extension management page
   - Extension popup

### 🔍 Additional Notes
- Theme preference is stored in localStorage
- Smooth transitions between themes
- Icons are optimized for different display densities
